### PR TITLE
docker-nydus-graphdriver: avoid `overlayfs: overlapping path'

### PR DESCRIPTION
**Pull request by hsiangkao**
For more details, see: https://lore.kernel.org/r/20190418144208.21832-1-amir73il@gmail.com

Signed-off-by: Gao Xiang \<hsiangkao@linux.alibaba.com\>

### CODE REVIEW

File: contrib/docker-nydus-graphdriver/plugin/nydus/nydus.go [hunk 1]
```diff
@@ -258,8 +258,10 @@ func (d *Driver) Get(id, mountLabel string) (fs containerfs.ContainerFS, retErr
 		return
 	}
 
+	newlowers := make([]string, 0)
 	for _, l := range lowers {
 		if l == id {
+			newlowers = append(newlowers, id)
```
Comment (changweige) [inline contrib/docker-nydus-graphdriver/plugin/nydus/nydus.go:264]: A preferred name convention would be `newLowers`

Comment (changweige) [reply contrib/docker-nydus-graphdriver/plugin/nydus/nydus.go:264]: `newlowers` as a slice forever accommodates only `id`?

**Comment by changweige** [comment]
So current overlayfs mount behavior is not defined per as https://lore.kernel.org/r/20190418144208.21832-1-amir73il@gmail.com ?
Can you help me understand how the overlapping happens around there?

**Review by imeoer — APPROVED** [review]

File: contrib/docker-nydus-graphdriver/plugin/nydus/nydus.go [hunk 2]
```diff
@@ -297,11 +299,11 @@ func (d *Driver) Get(id, mountLabel string) (fs containerfs.ContainerFS, retErr
 		// Relative path
 		nydusRelaMountpoint := path.Join(l, nydusDirName)
 		if _, err := os.Stat(path.Join(d.home, nydusRelaMountpoint)); err == nil {
-			lowers = append(lowers, nydusRelaMountpoint)
+			newLowers = append(newLowers, nydusRelaMountpoint)
```
Comment (bergwolf) [inline contrib/docker-nydus-graphdriver/plugin/nydus/nydus.go:302]: Indeed, overlapping path in lower dirs is broken by design with overlayfs.

**Review by bergwolf — APPROVED** [review]

**Pull request merged**